artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPSessionCallback.java

```diff
@@ -429,6 +442,12 @@ public void offerProducerCredit(final String address,
                                    final int threshold,
                                    final Receiver receiver) {
       try {
+         // TODO - Do something smart here with the paging bits to figure out the credit can be sent.
```

Comment:
I don't think there's anything to be done with paging. If this is an anonymous producer, there's nothing to be done with paging...

Either this is the proper implementation, hence you would need to remove the TODO, or this is something else broken.

I believe you should just remove the TODO here